src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/X500NameEncoder.cs

```diff
@@ -200,5 +252,445 @@ private static void AppendOid(StringBuilder decodedName, string oidValue)
 
             decodedName.Append('=');
         }
+
+        private enum ParseState
+        {
+            Invalid,
+            SeekTag,
+            SeekTagEnd,
+            SeekEquals,
+            SeekValueStart,
+            SeekValueEnd,
+            SeekEndQuote,
+            MaybeEndQuote,
+            SeekComma,
+        }
+
+        private static List<byte[][]> ParseDistinguishedName(
+            string stringForm,
+            List<char> dnSeparators,
+            bool noQuotes)
+        {
+            // 16 is way more RDNs than we should ever need. A fairly standard set of values is
+            // { E, CN, O, OU, L, S, C } = 7;
+            // The EV values add in
+            // {
+            //   STREET, PostalCode, SERIALNUMBER, 2.5.4.15,
+            //   1.3.6.1.4.1.311.60.2.1.2, 1.3.6.1.4.1.311.60.2.1.3
+            // } = 6
+            //
+            // 7 + 6 = 13, round up to the nearest power-of-two.
+            const int InitalRdnSize = 16;
+            List<byte[][]> encodedSets = new List<byte[][]>(InitalRdnSize);
+            char[] chars = stringForm.ToCharArray();
```

Comment:
I realize there's lots of other allocation happening, but my personal preference would be to avoid the extra allocation/copy and use unsafe with GetByteCount and ```char*``` rather than creating the ```char[]```.  I saw Morgan's comments about not using unsafe when there's no value, but here there seems to be value.